File: app/src/main/java/com/github/gfx/helium/di/AppComponent.java
package com.github.gfx.helium.di;

import com.github.gfx.helium.util.OkHttp3GlideModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {

    void inject(OkHttp3GlideModule okHttp3GlideModule);

    ActivityComponent plus(ActivityModule module);
}

File: app/src/main/java/com/github/gfx/helium/di/AppComponent.kt
package com.github.gfx.helium.di

import com.github.gfx.helium.StethoDelegator

import javax.inject.Singleton

import dagger.Component

@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {

    fun inject(stethoDelegator: StethoDelegator)

    operator fun plus(module: ActivityModule): ActivityComponent
}
